test/common/router/router_test.cc

```diff
@@ -178,6 +179,7 @@ TEST_F(RouterTest, UpdateServerNameFilterState) {
 TEST_F(RouterTest, UpdateSubjectAltNamesFilterState) {
   NiceMock<StreamInfo::MockStreamInfo> stream_info;
   auto dummy_option = absl::make_optional<envoy::config::core::v3::UpstreamHttpProtocolOptions>();
+  dummy_option.value().set_alt_header_name(":authority");
```

Comment:
In this file, instead of updating existing tests to set the new field to the value that would be used if the field wasn't set, add new test cases, so that both the case where the new field is unset, and the case where it is set, are handled. Add test coverage for cases when it is set to some other header, both for when that header is present in the request and absent in the request.

Reply:
Added more tests as per the suggestions.